pandas/tests/indexing/test_iloc.py

```diff
@@ -834,6 +835,32 @@ def test_iloc_setitem_dictionary_value(self):
         expected = DataFrame({"x": [1, 9], "y": [2, 99]})
         tm.assert_frame_equal(df, expected)
 
+    def test_iloc_setitem_conversion_to_datetime(self):
+        # GH#20511
+        df = DataFrame(
+            [["2015-01-01", "2016-01-01"], ["2016-01-01", "2015-01-01"]],
+            columns=["date0", "date1"],
+        )
+        df.iloc[:, [0]] = df.iloc[:, [0]].apply(
+            lambda x: to_datetime(x, errors="coerce")
+        )
+        expected = DataFrame(
+            {
+                "date0": [to_datetime("2015-01-01"), to_datetime("2016-01-01")],
```

Comment:
can you use Timestamp instead of to_datetime

Reply:
Done